CONTRIBUTING.md

```diff
@@ -0,0 +1,92 @@
+- Fork the repository on GitHub.
+- Clone onto your local development machine via `git clone https://github.com/{$GH-USERNAME}/operator-controller`.
+- Add the operator-framework/operator-controller as an upstream.
+  remote: `git remote add upstream https://github.com/operator-framework/operator-controller`.
+- Create a new branch from the default `main` branch and begin development.
+
+## Contribution flow
+
+Any new contribution should be accompanied by a new or existing issue. This issue can help track work, discuss the
+design and implementation, and help avoid wasted efforts of multiple people working on the same issue. Trivial changes,
+like fixing a typo in the documentation, do not require the creation of a new issue.
+
+### Small Contributions
+
+For simple contributions, this is a rough outline of what a contributor's workflow looks like:
+
+- Identify or create a GitHub Issue.
+- Create a topic branch from where to base the contribution. This is usually the main branch.
+- Make commits of logical units.
+- Make sure commit messages are in the proper format (see below).
+- Push changes in a topic branch to a personal fork of the repository.
+- Submit a pull request to the operator-framework/operator-controller repository.
+- Wait and respond to feedback from the maintainers listed in the OWNERS file.
+
+### Large Contributions
+
+The Operator Controller project is an open source project featuring contributors from many companies and backgrounds.
+In an effort to coordinate ongoing efforts from multiple contributors, large scale features are tracked on the
+[OLM V1 GitHub Project](https://github.com/orgs/operator-framework/projects/8/views/14?pane=info) in a `Milestone #` tab.
+Unassigned tickets in a `Milestone #` are available for contributors to take, simply assign the ticket to yourself and
+reach out to the community by commenting on the issue or by messaging us on the [#olm-dev](https://kubernetes.slack.com/archives/C0181L6JYQ2) Slack Channel.
+
+If you are interested in proposing a "focus" for a milestone, the contribution workflow should be similar to the following:
+
+- Create a [GitHub Issue](https://github.com/operator-framework/operator-controller/issues/new), labeled with "milestone-proposal", that includes a link to a [HackMD](https://hackmd.io) capturing the proposed changes.
+- Add an item to the [OLM Working Group and Issue Triage Meeting Agenda](https://docs.google.com/document/d/1Zuv-BoNFSwj10_zXPfaS9LWUQUCak2c8l48d0-AhpBw/edit) to introduce your design to the community.
+- Introduce your proposed changes at the weekly [OLM Working Group Meeting](https://github.com/operator-framework/community#operator-lifecycle-manager-working-group) 
+
+The community will then discuss the proposal, consider the benefits and cost of the feature, iterate on the design, and decide if it project should pursue the design. If accepted, the design will be queued up for a subsequent milestone.
+
+Thanks for contributing!
+
+### Code Review
```

Comment:
Something that might be nice to add here is some guidelines on when and how to reach out to maintainers directly about PRs. For example, is it expected that my PR may take 2 weeks to get a review? If I haven't received a review in a week, where & who should I reach out to to make sure my changes are getting reviewed?

Reply:
Good point! We may want to add an OWNERS or MAINTAINERS file and then simply mention to tag the individuals in that group.

Reply:
I've created [an issue](https://github.com/operator-framework/operator-controller/issues/153) to track this work